### PR TITLE
Fix old generation dexsearching by move.

### PR DESCRIPTION
**Pull request by SSlayer9**
Potential fix to https://www.smogon.com/forums/threads/bug-reports-v4-read-original-post-before-posting.3663703/post-8469484.

Untested and will probably fail.

### CODE REVIEW

File: server/chat-plugins/datasearch.ts [hunk 1]
```diff
@@ -538,6 +538,10 @@ function runDexsearch(target: string, cmd: string, canAll: boolean, message: str
 			if (targetMove.exists) {
 				const invalid = validParameter("moves", targetMove.id, isNotSearch, target);
 				if (invalid) return {error: invalid};
+				if (!isNaN(parseInt(cmd.replace('ds', '')))) {
+					 let gen = partseInt(cmd.replace('ds', ''));
```
Comment (mia-pi-git) [inline server/chat-plugins/datasearch.ts:542]:
```suggestion
					 let gen = parseInt(cmd.replace('ds', ''));
```

**Comment by KrisXV** [comment]
Why not just move the maxgen check above the other checks?

**Comment by mia-pi-git** [comment]
This code doesn't work yet - you should probably change this to a draft PR for now.

**Comment by AnnikaCodes** [comment]
> Why not just move the maxgen check above the other checks? That way, you can just do `if (move.gen > dex.gen)`

This only works if the `maxgen` parameter is parsed before the move, which is not the case when using `/ds(gen #)`.
The way I'd handle this is to compare the move's generation and `maxGen` later on, when all the parameters have been parsed and the search is actually being run. I have tested this and confirmed it works.

**Comment by SSlayer9** [comment]
Left to annika

**Pull request closed without merging**